src/XMakeBuildEngine/Evaluation/Evaluator.cs

```diff
@@ -816,11 +816,41 @@ private static ProjectTargetInstance ReadNewTargetElement(ProjectTargetElement t
             string endPass2 = String.Format(CultureInfo.CurrentCulture, "Evaluate Project {0} - End Pass 2 (Item Definitions)", projectFile);
             DataCollection.CommentMarkProfile(8818, endPass2);
 #endif
+            LazyItemEvaluator<P, I, M, D> lazyEvaluator = null;
+            lazyEvaluator = new LazyItemEvaluator<P, I, M, D>(_data, _itemFactory, _buildEventContext, _loggingService);
+
             // Pass3: evaluate project items
             foreach (ProjectItemGroupElement itemGroupElement in _itemGroupElements)
             {
-                EvaluateItemGroupElement(itemGroupElement);
+                EvaluateItemGroupElement(itemGroupElement, lazyEvaluator);
+            }
+
+            if (lazyEvaluator != null)
+            {
+                IList<LazyItemEvaluator<P, I, M, D>.ItemData> items = lazyEvaluator.GetAllItems();
+                foreach (var itemData in items)
+                {
+                    if (itemData.ConditionResult)
+                    {
+                        _data.AddItem(itemData.Item);
+
+                        if (_data.ShouldEvaluateForDesignTime)
+                        {
+                            _data.AddToAllEvaluatedItemsList(itemData.Item);
+                        }
+                    }
+                }
+
+                if (_data.ShouldEvaluateForDesignTime)
+                {
+                    foreach (var itemData in items)
+                    {
+                        _data.AddItemIgnoringCondition(itemData.Item);
```

Comment:
It was trying to emulate [this ](https://github.com/Microsoft/msbuild/blob/xplat/src/XMakeBuildEngine/Evaluation/Evaluator.cs#L1788-L1807)code, but it can actually be removed in this case since the structure changed a bit.